blockchain-albatross/src/blockchain.rs

```diff
@@ -592,6 +599,72 @@ impl Blockchain {
                 return Err(PushError::InvalidBlock(BlockError::InvalidJustification));
             }
 
+            // Check if there are two blocks in the same slot. Since we already
+            // verified the validator for the current slot, this is enough to
+            // check for fork proofs. A fork proof doesn't apply to the genesis
+            // block, so skip block 0.
+            if block.block_number() != 0 {
+                // Count the microblocks after the last macroblock
+                let count = block.block_number() - policy::macro_block_before(block.block_number());
+                let microblocks: Vec<Block> = self.chain_store.get_blocks_backward(block.parent_hash(), count - 1, false, Some(&read_txn));
```

Comment:
Fork blocks may only occur at the same height and view-number.
More specifically, a fork proof consists of two blocks:
- that are different (i.e., the content and thus the hash is different)
- at the same height
- with the same view number
- from the same slot (i.e., signed by the same validator)

So, we only need to get all micro blocks at the height `block.block_number()`, check if they have the same view number and are signed by the same person.